### ISSUE TITLE
Infinite loop when durandal observable plugin is enabled

### ISSUE THREAD

**Issue by navels** [kyeotic/durandal-grid#4]
If you enable the observable plugin in durandal _and_ you have a table for which you specify column config _and_ there are no searchable columns, knockout gets into an infinite loop when the table is first rendered.

For example, this kind of grid config will do the trick:

```
  gridConfig: {
    data: [ { firstName : 'Joe' } ],
    columns : [
      { header : 'First Name', property : 'firstName' }
    ]
  }
```

with the bare-bones HTML:

```
<table class="paging-container grid-table" data-bind="grid: gridConfig"></table>
```

Just watch the console and you should see these scrolling by:

```
Binding widgets/grid/view GridWidget
```

If you add "canSearch : true" to the column config, the problem does not occur.

ps great work, thanks for putting this out there!

**Comment by navels** [comment]
I think I see the fix.  In the definition of filteredRows, we should return a copy of the rows array if there are no search columns:

```
if(self.searchColumns().length == 0)
  return rows.slice();
```

instead of returning just rows.

**Comment by kyeotic** [comment]
fixed with 6850a85
